src/io/std_file_descriptor.cr

```diff
@@ -0,0 +1,26 @@
+require "./syscall"
+require "./file_descriptor"
+
+class IO::StdFileDescriptor < IO::FileDescriptor
```

Comment:
I would probably avoid a subclass and just add a boolean in the constructor to do this small logic. But I'd like to know others' opinions.

Reply:
There was more code when I started out, trying to dup old FDs and stuff. But yeah this could be put straight into IO::FD I guess.

Reply:
You don't need a flag, just a differently named constructor, like `FileDescriptor.standard` which calls `FileDescriptor.new` with the newly opened fd.

Reply:
We don't need to test for `tty?` all the time either. A special constructor is a good idea.